frontend/public/components/catalog-items.jsx

```diff
@@ -1,48 +1,201 @@
+        showAllItemsForCategory: null,
+      });
+      return;
+    }
+
+    const { currentCategories } = this.state;
+    const categories = parentCategory ? currentCategories : allCategories;
+    const activeCategory = _.find(categories, { id: category });
+    if (!activeCategory) {
+      return;
+    }
+
+    const { numItems, subcategories } = activeCategory;
+    const state = {
+      activeTabs: parentCategory ? [parentCategory, category] : [category],
+      numItems,
+    };
+    if (_.isEmpty(subcategories)) {
+      // no sub-categories, show all items for selected category
+      _.assign(state, {
+        currentCategories: categories,
+        showAllItemsForCategory: category,
+      });
+    } else {
+      // show list of sub-categories
+      _.assign(state, {
+        currentCategories: subcategories,
+        showAllItemsForCategory: null,
+      });
+    }
+    this.setState(state);
+  }
+
+  onActivateTab(tabs) {
+    const category = _.last(tabs);
+    const parent = tabs.length > 1 ? _.first(tabs) : null;
+    this.syncTabsAndTiles(category, parent);
+  }
+
+  renderCategoryTiles(category) {
+    const { showAllItemsForCategory } = this.state;
+    const { id, label, parentCategory, items } = category;
+    if (showAllItemsForCategory && id !== showAllItemsForCategory) {
+      return null;
+    }
+
+    return <CatalogTileView.Category
+      key={id}
+      title={label}
+      totalItems={items && category.items.length}
+      viewAll={showAllItemsForCategory === id}
+      onViewAll={() => this.syncTabsAndTiles(id, parentCategory)}>
+      {_.map(items, ((item) => {
+        const { obj, tileName, tileImgUrl, tileIconClass, tileProvider, tileDescription, href } = item;
+        const uid = obj.metadata.uid;
+        const iconClass = tileIconClass ? `icon ${normalizeIconClass(tileIconClass)}` : null;
+        const vendor = tileProvider ? `Provided by ${tileProvider}` : null;
+        const catalogTile = <CatalogTile
+          id={uid}
+          key={uid}
+          title={tileName}
+          iconImg={tileImgUrl}
+          iconClass={iconClass}
+          vendor={vendor}
+          description={tileDescription} />;
+        return href ? <Link key={uid} className="co-catalog-item-tile" to={href}>{catalogTile}</Link> : catalogTile;
+      }))}
+    </CatalogTileView.Category>;
+  }
+
+  getCategoryLabel(categoryID) {
+    const { allCategories } = this.state;
+    return _.find(allCategories, { id: categoryID }).label;
+  }
 
-export const CatalogList = ({items}) =>
-  <CatalogTileView>
-    <CatalogTileView.Category key="all" title="All">
-      {_.map(items, (item) => <CatalogItem key={item.obj.metadata.uid} item={item} />)}
-    </CatalogTileView.Category>
-  </CatalogTileView>;
+  render() {
+    const { activeTabs, showAllItemsForCategory, currentCategories, numItems } = this.state;
+    const activeCategory = showAllItemsForCategory ? _.find(currentCategories, { id: showAllItemsForCategory }) : null;
```

Comment:
I'm not sure the `_.find` check is right here... `state.showAllItemsForCategory` is the category object itself, right?

(This might be my bug...)

Reply:
no `showAllItemsForCategory` is the id of a category :-)